src/components/pages/dashboard-pages/AddClass.jsx

```diff
@@ -64,91 +70,93 @@ const AddClass = () => {
     }));
   };
 
-  const createMultipartForm = (data) => {
-    const multipartForm = new FormData();
-
-    for (let key in data) {
-      if (key === 'ages') {
-        multipartForm.append('ages[minAge]', data.ages.minAge);
-        multipartForm.append('ages[maxAge]', data.ages.maxAge);
-      } else if (key === 'availableTime') {
-        multipartForm.append(
-          'availableTime[date]',
-          new Date(data.availableTime.date)
-        );
-        multipartForm.append(
-          'availableTime[startTime]',
-          data.availableTime.startTime
-        );
-      } else if (data[key] !== null && data[key] !== undefined) {
-        multipartForm.append(key, data[key]);
-      }
+const handleSubmit = async (e) => {
+  e.preventDefault();
+  let availableTimeError = false;
+  formData.availableTime.forEach((time) => {
```

Comment:
Hi Valya! Here is very common mistake, you just need to remember that forEach is synchronous function and you shouldn't use it in async function.  Here the documentations:
[forEach() expects a synchronous function — it does not wait for promises. Make sure you are aware of the implications while using promises (or async functions) as forEach callbacks.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach)

example what to choose instead of forEach: https://gist.github.com/joeytwiddle/37d2085425c049629b80956d3c618971

Technically it can recalculate fast and finish calculations before the promice returned, but if not - you'll not receive all data